test/assert.js

```diff
@@ -74,6 +74,26 @@ assert.holdsItem = function (pokemon, message) {
 	});
 };
 
+assert.trapped = function (fn) {
+	assert.throws(() => fn(), Error, "[Invalid choice] Can't switch: The active Pokémon is trapped");
+};
+
+assert.cantMove = function (fn, pokemon, move) {
+	if (pokemon && move) {
+		assert.throws(() => fn(), Error, new RegExp(`\\[Invalid choice\\] Can't move:.*${pokemon}.*${move}`, 'i'));
+	} else {
+		assert.throws(() => fn(), Error, /\[Invalid choice\] Can't move:/);
+	}
+};
+
+assert.cantUndo = function (fn) {
+	assert.throws(() => fn(), Error, /\[Invalid choice\] Can't undo:/);
+};
```

Comment:
Uhm, actually... I have just checked the ``assert`` API at https://nodejs.org/api/assert.html#assert_assert_throws_fn_error_message:

> assert.throws(fn[, error][, message])
fn <Function>
error <RegExp> | <Function> | <Object> | <Error>
message <string>

[...]

If specified, error can be a Class, RegExp [...]

If specified, message will be appended to the message provided by the AssertionError if the fn call fails to throw or in case the error validation fails.

So, from above, ``error`` should be a ``RegExp`` matching the choice error, and ``message`` should be something like ``'Choice unexpectedly accepted``

Reply:
Per above, ``error`` should be a ``RegExp`` matching the choice error, and ``message`` should be something like ``'Choice unexpectedly accepted``

Reply:
Ah, so right now its just matching the `Error` constructor, so its basically just a vanilla `assert.throws` with a weird regexp error message? Thanks for being so 'flexible', JS.

Reply:
Switched things around, but `message` still doesn't seem to be working correctly. Will dig in further.

Reply:
OK, I'm dumb. `message` is working, it just doesn't work the way I thought it would. I thought it would always be used if the test fails, but its only used if the *assertion* fails (ie/ *no* error is thrown). If the test fails because an error is thrown, you see the *exception*'s message. This makes sense and is fine, I just didn't understand what was going on and couldn't verify my `message`s were working correctly until I figured it out.

I think I'm done with this change. I'd be happy to submit this as is and let you take over the remaining cases if you're still up for it. :)

Reply:
Great! Oh, oops! Sorry for the merge conflict :[

Reply:
No worries.